### PR TITLE
Added callback calls on CallAsync service errors

### PR DESCRIPTION
**Pull request by brakmic-aleksandar**
**Pull request type**

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

**What is the current behavior?**
When client calls service which doesn't exist/isn't initiated yet it fails silently.

Issue Number: #371

**What is the new behavior?**
Client will let user know that it couldn't connect to service through callback.

**Does this introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by FlorianReimold** [comment]
This adds a new API function right? Wich would mean it breaks ABI compatibility :/

**Comment by brakmic-aleksandar** [comment]
@FlorianReimold It adds private function to impl class which should prevent breakage of ABI compatibility, if i'm not mistaken.

**Comment by rex-schilasky** [comment]
For the new ` CServiceClientImpl::CallAsync` methods I would prefer the return type `bool` too. Like there counterpart signatures `bool CServiceClientImpl::Call`.

File: ecal/core/src/service/ecal_service_client_impl.cpp [hunk 1]
```diff
@@ -147,13 +147,23 @@ namespace eCAL
 
   void CServiceClientImpl::CallAsync(const std::string& host_name_, const std::string& method_name_, const std::string& request_)
```
Comment (rex-schilasky) [inline ecal/core/src/service/ecal_service_client_impl.cpp:148]: `bool CServiceClientImpl::CallAsync`

**Review by rex-schilasky — CHANGES_REQUESTED** [review]

**Pull request merged**